core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2098,31 +2100,37 @@ class Log(@volatile var dir: File,
     logString.toString
   }
 
+  private def removeAndDeleteSegments(segments: List[LogSegment], asyncDelete: Boolean = true): Unit = {
```

Comment:
Instead of List[LogSegment] use Iterable[LogSegment]) ?

Reply:
I think using `List` is safer, to force callers to materialize the view of segments it derived from the `segments` map.

Reply:
I am nit sure I understand how using List makes it safer, one could really has a list or a array, it should not matter

Reply:
This is not a public API so I am not sure if the goal is to provide a generic API. Like I said, the goal is to materialize any iterators derived from the underlying collection, so we end up mutating the collection while using the iterators. I could change to taking in an `Iterable` and then convert it to a list inside `removeAndDeleteSegments` if you think that is better.

Reply:
not needed, that would be cosmetic, my comment was on conversion of iterable to list when the underlying function is essentially iterating to delete the unordered members in the set